### ISSUE TITLE
tfl.while and tfl.if legalization to tosa

### ISSUE THREAD

**Issue by scottcantreid** [iree-org/iree#9863]
### Issue body

I will be writing a pass to convert `tfl.while`/`tfl.if` to `tosa.while`/`tosa.if`, respectively.

I'm taking inspiration from the implementation of the [TOSA->SCF conversion](https://github.com/llvm/llvm-project/blob/main/mlir/lib/Conversion/TosaToSCF/TosaToSCF.cpp). When completed, this pass should also lower all `tfl.yield` operations to `tosa.yield`.

As an external contributor, I will contribute this pass locally to `iree_tf_compiler/TFL`. Ultimately, this probably belongs within the tensorflow repository. @rsuderman has kindly agreed to help migrate this into tensorflow.

**Comment by alliemartens** [comment]
Thanks, Scott for filing this! 
@rsuderman Please help add priority to this issue.

**Comment by scottcantreid** [comment]
I believe I have a working pass for `while`, but I want to clarify the signature for `if` before I write that one.

In the `tfl` dialect, `IfOp` has only a single argument: [`TFL_BoolTensor:$cond`](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/mlir/lite/ir/tfl_ops.td#L4966)

However, in the `tosa` dialect, `IfOp` has the an equivalent boolean argument, as well as a [`Variadic<Tosa_Tensor>:$inputs`](https://github.com/llvm/llvm-project/blob/main/mlir/include/mlir/Dialect/Tosa/IR/TosaOps.td#L1836-L1839). Does anyone know the purpose of this variadic argument?

**Comment by jpienaar** [comment]
I'd expect this to be used as form before export given spec form https://www.mlplatform.org/tosa/tosa_spec.html#_cond_if, but ops aren't isolated from above so shouldn't be needed in general AFAICS

**Comment by scottcantreid** [comment]
Thanks! So my assumption is that a tensor needs to be passed into `input_list` in order to be in scope in the `then/else` regions. This is different from `scf.if` and `tfl.if`, where any variable previously defined in the graph is in scope.

**Comment by scottcantreid** [comment]
> but ops aren't isolated from above so shouldn't be needed in general AFAICS
Given this, I can pretty easily write a lowering that leaves the `input_list` empty. I wonder if the TOSA group would be open to changing the spec to remove this unnecessary argument.

**Comment by jpienaar** [comment]
Yes, so I think I'd do what you propose and then we could have (like we do for TFLite) a "prepare for export" pass that makes it args. But for all optimizations, I'd prefer it in region form. And j don't think for iree it matters, only place that seems like it could matter is for export to interpreter/TOSA flatbuffer
